### PR TITLE
add TensorFlow TFRecord file support #382

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

**Comment by nevillelyh** [comment]
@ravwojdyla thoughts?

File: scio-core/src/main/scala/com/spotify/scio/io/TFRecord.scala [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    case object NONE extends CompressionType
+    case object ZLIB extends CompressionType
+    case object GZIP extends CompressionType
+  }
+
+  /** Default read options. */
+  val readDefault = TFRecordOptions(CompressionType.AUTO)
+
+  /** Default write options. */
+  val writeDefault = TFRecordOptions(CompressionType.NONE)
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/io/TFRecord.scala]: Should it be compressed by default?

Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/io/TFRecord.scala]: Not sure since the Python TF version defaults to NONE.

**Comment by nevillelyh** [comment]
It reads raw/zlib/gzip files generated by Python TF. I can look into TF but it'll be a lot more work like including a TF distribution in the code base or checking in pre-generated binary file.

**Pull request merged**